src/components/listings/ImageGridItem.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react'
+import { ImageBackground, Pressable } from 'react-native'
+import Animated, {
+  useSharedValue,
+  withDelay,
+  withTiming,
+} from 'react-native-reanimated'
+import Icon from '@/components/Icon'
+import { useColours } from '@/hooks/useTailwind'
+import { IconSize } from '@/types/icon'
+import { ImageGridItemProps } from '@/types/imageGrid'
+
+const DURATION = 200
+const DELAY = 50
+
+function ImageGridItem(props: ImageGridItemProps) {
+  const { image, handleSelectedImages, isSelected } = props
+
+  const colours = useColours()
+
+  const [selected, setSelected] = useState(isSelected)
+
+  const opacity = useSharedValue(0)
+
+  function handleSelected(imageId: number) {
+    handleSelectedImages(imageId)
+
+    if (selected) {
+      opacity.value = withDelay(DELAY, withTiming(0, { duration: DURATION }))
+    } else {
+      opacity.value = withDelay(DELAY, withTiming(1, { duration: DURATION }))
+    }
+
+    setSelected(!selected)
```

Comment:
do we need to keep a record of this internally or can we just use `isSelected` from props?